core/src/apps/bitcoin/sign_tx/zcash.py

```diff
@@ -34,32 +34,23 @@
 OVERWINTERED = const(0x80000000)
 
 
-class Overwintered(Bitcoinlike):
+class Zcashlike(Bitcoinlike):
     def __init__(self, tx: SignTx, keychain: Keychain, coin: CoinInfo) -> None:
         ensure(coin.overwintered)
         super().__init__(tx, keychain, coin)
 
-        if self.tx.version == 3:
-            if not self.tx.branch_id:
-                self.tx.branch_id = 0x5BA81B19  # Overwinter
-        elif self.tx.version == 4:
-            if not self.tx.branch_id:
-                self.tx.branch_id = 0x76B809BB  # Sapling
-        else:
+        if self.tx.version != 4 or self.tx.branch_id is None:
```

Comment:
let's enforce `branch_id` in `sanitize_sign_tx` instead of checking here

Reply:
Fixed 016af5708c5b420cb6973e6f1593aa16e151b45b.

Reply:
no, no, keep the `tx.version` check here.

there's this kinda nebulous split of "data sanitization" (helpers.sanitize_*) vs "business logic" (the signer class) that we're trying to uphold, and knowing that we can't sign any other versions goes on the "business logic" IMHO

Reply:
So `branch_id` in `sanitize_sign_tx` and leave `tx.version` here? PTAL @ b0c672de7acdcfbfcfaee011ac000d07723d93df